src/openssl/hmac.cr

```diff
@@ -1,6 +1,23 @@
 require "./lib_crypto"
 
+# Allows computing Hash-based Message Authentication Code (HMAC).
+#
+# It is a type of message authentication code (MAC)
+# involving a hash function in combination with a key.
+#
+# HMAC can be used to verify the integrity of a message as well as the authenticity.
 class OpenSSL::HMAC
+  # Returns the *data* in combination with the *key* hash computed with *algorithm* Digest.
```

Comment:
I think this would be written better simplified:
> Returns the HMAC digest of *data* using the secret key *key*.

Adding a note that the result is binary might be useful.
>It may contain non-ASCII bytes, including NUL bytes.